village/src/App.js

```diff
@@ -14,11 +18,60 @@ class App extends Component {
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
+
+
+  //I made this to hand down to the SmurfForm component, which will allow it to update the app state
+  //Needs explicit binding to work
+  handleSmurfsUpdate(smurfs) {
+    this.setState({smurfs: smurfs});
+  }
+
+  //Use a delete request to delete smurfs.  Gets passed all the way down to the Smurf component
+  handleSmurfDelete(id) {
+    axios.delete(`http://localhost:3333/smurfs/${id}`)
+      .then((response) => this.handleSmurfsUpdate(response.data))
+      .catch((err) => console.log(err));
+
+  }
+
+
+  componentDidMount() {
+    //get the smurf data from the server
+    //then put it on the state
+    axios.get('http://localhost:3333/smurfs')
+      .then((response) => this.handleSmurfsUpdate(response.data))
+      .catch((err) => console.log(err));
+  }
+
+
+
   render() {
     return (
+
       <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
+        <Route path="/smurfs" render={(props) => (
+          <div>
+            <SmurfForm handleSmurfsUpdate={this.handleSmurfsUpdate.bind(this)} />
```

Comment:
Using the arrow function allows you to not have to explicitly bind this, but still great job